index/scorch/snapshot_index_vr.go

```diff
@@ -50,7 +51,19 @@ type IndexSnapshotVectorReader struct {
 	currID        index.IndexInternalID
 	ctx           context.Context
 
-	searchParams json.RawMessage
+	searchParams     json.RawMessage
+	eligibleDocIDs   []index.IndexInternalID
+	requireFiltering bool
+}
+
+func (i *IndexSnapshotVectorReader) EligibleDocIDs() *roaring.Bitmap {
+	res := roaring.NewBitmap()
```

Comment:
Need to re-think whether this really needs to be a roaring bitmap.

Reply:
Having this as a roaring bitmap simplifies the AND'ing with the global doc nums in each segment.

Reply:
done, pls take a look.